src/software/ai/hl/stp/play/direct_free_kick_play.cpp

```diff
@@ -39,19 +39,19 @@ bool IndirectFreeKickPlay::isApplicable(const World &world) const
            min_dist_to_corner >= CornerKickPlay::BALL_IN_CORNER_RADIUS;
 }
 
-bool IndirectFreeKickPlay::invariantHolds(const World &world) const
+bool DirectFreeKickPlay::invariantHolds(const World &world) const
 {
     return (world.gameState().isPlaying() || world.gameState().isReadyState()) &&
            (!Evaluation::teamHasPossession(world, world.enemyTeam()) ||
             Evaluation::teamPassInProgress(world, world.friendlyTeam()));
```

Comment:
@MathewMacDougall - was the "indirect free kick" `invariantHolds` and `isApplicable` already general enough to encompass the direct and indirect free kicks? Or do we need changes here as well? Seems like they are, but just want to explicitly sanity check this.

Reply:
The `invariantHolds` should be fine for both.

For `isApplicable`, we should probably remove the condition that checks `min_dist_to_corner >= CornerKickPlay::BALL_IN_CORNER_RADIUS` since neither play is now explicitly a "corner" kick. This might be easiest to just remove for now but leave a ticket to revisit it, since this should be simplified by incoming rule changes (no more indirect kicks, only direct kicks)

Reply:
I think we should keep that, it's supposed to prevent this play from being applied when we _are_ in a corner kick scenario (which has a different play).

Reply:
Gotcha. Should be all good then